Scale CENNZ and CPAY down by 10^12 factor on a runtime upgrade

aliXsed opened the pull request:
Uses the sudo key to burn the balances.

Closes #228.

Commit:
Scale CENNZ and CPAY down by 10^12 factor on a runtime upgrade

Commit:
Bring the whole logic from plug to cennznet

jordy25519 commented on runtime/src/lib.rs:
```diff
@@ -815,6 +816,24 @@ impl_runtime_apis! {
+		scale_down(GenericAsset::spending_asset_id());
+		scale_down(GenericAsset::staking_asset_id());
+
+		0 // No weight
```
`Zero::zero()`

aliXsed replied:
Done

jordy25519 commented on runtime/tests/tests.rs:
```diff
@@ -13,21 +13,21 @@
 	Babe, Call, CennzxSpot, CheckedExtrinsic, ContractTransactionBaseFee, EpochDuration, Event, Executive,
 	GenericAsset, Header, Origin, Runtime, Session, SessionsPerEra, Staking, System, Timestamp, TransactionBaseFee,
-	TransactionByteFee, TransactionPayment, UncheckedExtrinsic,
+    TransactionByteFee, TransactionPayment, UncheckedExtrinsic, ScaleDownFactor,
```
Did you run `cargo fmt`?

aliXsed replied:
Yes, and I'm confused. Is a tab 4 spaces or a tab?

jordy25519 replied:
a tab char

aliXsed replied:
We need to add .rustfmt.toml to the project to specify our own formatting rather than the defaults. I'm afraid the defaults might change with different versions of rustfmt.

aliXsed replied:
Okay we actually have this file but somehow not working for me

aliXsed replied:
Will format the files again

aliXsed replied:
Done. The problem was with CLION doing its own formatting.

jordy25519 reviewed (commented):
lgtm! will approve once CI passes and tested on a dev chain. Not sure how implementing `on_runtime_upgrade` on the `Runtime` struct will actually play out in practice vs. for a module

Commit:
Fix formatting

aliXsed commented:
> module

@holygits good point. Let me investigate that.

Commit:
Move runtime upgrade logic to a new minimal and temporary module

When a runtime upgrade happens, it's the on_runtime_upgrade of the modules that would be called and so we need to have a module for our purpose and add it to Runtime

KarishmaBothara commented on runtime/tests/mock.rs:
```diff
@@ -94,6 +98,10 @@ impl ExtBuilder {
 		self.validator_count = count;
 		self
 	}
+	pub fn sudoer(mut self, key: AccountId32) -> Self {
```
Doubt here, do we need changes for scaling in tests/mock.rs too? As we now have separate module and its mock contains all these related function..

aliXsed replied:
Yup, you are right. we don't need this anymore here. Thank you for seeing it.

Commit:
Remove the old test

Commit:
Remove all the vestiges of the old code in runtime
